File: Python/PrefixSum.py
INF=10**9+7

def PrefixSum():
    for i in range(0, N):
        sum[i+1]=arr[i]+sum[i]

N, M=map(int, input().split())
arr=list(map(int, input().split()))
sum=[0]*(N+1)
ans=-INF
PrefixSum()
for i in range(N-M+1):
    ans=max(ans,sum[i+M]-sum[i])
print(ans)

File: C++/PrefixSum.cpp
#include <bits/stdc++.h>
#define ll long long
#define INF (1e9+7)
using namespace std;
ll N, M, ans=-INF;
ll arr[1005], sum[1005];
void PrefixSum(){
    for (int i=1;i<=N;i++)
        sum[i]=arr[i-1]+sum[i-1];
}
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin >> N >> M;
    for (int i=0;i<N;i++)
        cin >> arr[i];
    PrefixSum();
    for (int i=0;i<=N-M;i++)
        ans=max(ans,sum[i+M]-sum[i]);
    cout << ans;
    return 0;
}